File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/oauth/jwt/JwtConfiguration.java
package io.twogether.nbe_5_7_2_02team.oauth.jwt;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import org.springframework.boot.context.properties.ConfigurationProperties;

@Getter
@RequiredArgsConstructor
@ConfigurationProperties(prefix = "custom.jwt")
public class JwtConfiguration {

    private final Validation validation;
    private final Secrets secrets;

    @Getter
    @RequiredArgsConstructor
    public static class Validation {
        private final Long access;
        private final Long refresh;
    }

    @Getter
    @RequiredArgsConstructor
    public static class Secrets {
        private final String originKey;
        private final String appKey;
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/oauth/jwt/JwtConfiguration.kt
package io.twogether.nbe_5_7_2_02team.oauth.jwt

import lombok.Getter
import lombok.RequiredArgsConstructor
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "custom.jwt")
class JwtConfiguration(
    val validation: Validation,
    val secrets: Secrets
) {

    data class Validation(
        val access: Long,
        val refresh: Long
    )

    data class Secrets(
        val originKey: String,
        val appKey: String
    )
}
